source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -442,12 +465,18 @@ static void *ns_mem_internal_alloc(ns_mem_book_t *book, const ns_mem_block_size_
     block_ptr[1 + data_size] = data_size;
 
 done:
-    if (book->mem_stat_info_ptr) {
-        if (block_ptr) {
-            //Update Allocate OK
+
+
+    if (block_ptr) {
+        //Update Allocate OK
+        book->heap_alloc_total_bytes += (data_size + 2) * sizeof(ns_mem_word_size_t);
+        if (book->mem_stat_info_ptr) {
```

Comment:
How about having the null check in the dev_stat_update function to reduce code size

Reply:
Good point is it eating performance then. It is some kind of balancing.